### ISSUE TITLE
parent and sibling initializers in the arms of a conditional

### ISSUE THREAD

**Issue by cassella** [chapel-lang/chapel#5960]
### Summary of Problem

```
inaninit.chpl:8: In initializer:
inaninit.chpl:12: error: field initialization not allowed before this.init()
```

When the field initialization is not before this.init():

### Steps to Reproduce

**Source Code:**

```chapel
class Foo {
  var a: int = 42;

  proc init(s: string) {
    a = s.length;
  }
  
  proc init(val: uint) {
    if (val < 5) {
      this.init("foo");
    } else {
      a = val;      // line 12
      super.init();
    }
  }
}

proc main() {
  var c = new Foo(2);
  writeln(c.a);
  delete c;

  c = new Foo(10);
  writeln(c.a);
  delete c;
}
```

I'd expected the output

```
3
10
```

Or, failing that, an error about being unable to call a sibling
initializer in one arm of a conditional and a parent initializer in
the other.  But without the a = val line, it compiles and produces the
output

```
3
42
```

**Compile command:**

```chpl inaninit.chpl -o inaninit```

**Execution command:**

```./inaninit -nl 1```

**Associated Future Test(s):**

``test/classes/initializers/cond-both-1-plus-field-init.chpl``

### Configuration Information

```
fortytwo@magrathea:~/src/chapel (master)$ chpl --version
chpl Version 1.15.0.a5fb8bb
Copyright (c) 2004-2017, Cray Inc.  (See LICENSE file for more details)
fortytwo@magrathea:~/src/chapel (master)$ printchplenv --anonymize
CHPL_TARGET_PLATFORM: linux64
CHPL_TARGET_COMPILER: gnu
CHPL_TARGET_ARCH: native *
CHPL_LOCALE_MODEL: flat
CHPL_COMM: gasnet *
  CHPL_COMM_SUBSTRATE: udp
  CHPL_GASNET_SEGMENT: everything
CHPL_TASKS: qthreads
CHPL_LAUNCHER: amudprun
CHPL_TIMERS: generic
CHPL_UNWIND: none
CHPL_MEM: jemalloc
CHPL_MAKE: make
CHPL_ATOMICS: intrinsics
  CHPL_NETWORK_ATOMICS: none
CHPL_GMP: gmp
CHPL_HWLOC: hwloc
CHPL_REGEXP: re2
CHPL_WIDE_POINTERS: struct
CHPL_AUX_FILESYS: none
fortytwo@magrathea:~/src/chapel (master)$ gcc --version
gcc (Ubuntu 5.4.0-6ubuntu1~16.04.4) 5.4.0 20160609
```

**Comment by lydia-duncan** [comment]
This is supposed to work.  I am surprised that my tests of the if branches did not handle this case, nor that @noakesmichael's testing did not notice it as well.  Thanks for spotting!

**Comment by lydia-duncan** [comment]
This issue was resolved in #6095
